pkg/plugin/resources.go

```diff
@@ -254,6 +275,133 @@ func (app *App) handleVectorSearch(w http.ResponseWriter, req *http.Request) {
 	w.Write(bodyJSON)
 }
 
+// SaveLLMOptInDataToGrafanaCom persists Grafana-managed LLM opt-in data to grafana.com.
+// This is required because provisioned plugins' settings do not survive a restart, and are
+// always reset to the state in grafana.com's provisioned-plugins endpoint.
+//
+// This function (and getPluginID) use the [/api/gnet][] endpoint, which [proxies requests][] to
+// the Grafana instance's configured [`GrafanaComUrl` setting][], which is set to the correct
+// value for the environment in Hosted Grafana instances.
+//
+// [/api/gnet]: https://github.com/grafana/grafana/blob/4d8287b319514b750617c20c130ffc424a3ecf2c/pkg/api/api.go#L677
+// [proxies requests]: https://github.com/grafana/grafana/blob/4bc582570ef7e713599ab3f2009fa75c27bb8a02/pkg/api/grafana_com_proxy.go#L28
+// [`GrafanaComUrl` setting]: https://github.com/grafana/grafana/blob/460be702619428e455ba74f8fb3bb563c1bea43a/pkg/setting/setting.go#L1088
+// Handles requests to /save-llm-state endpoint, and pushes state to GCom.
+func (app *App) handleSaveLLMState(w http.ResponseWriter, req *http.Request) {
+	log.DefaultLogger.Debug("Handling request to save LLM state to gcom..")
+	if app.saToken == "" {
+		// not available in Grafana < 10.2.3 or if externalServiceAccounts feature flag is not enabled
+		log.DefaultLogger.Warn("Service account token not available; cannot save LLM state to gcom")
+		return
```

Comment:
This should handle an error as well I think?

Reply:
In this case we want to return without an error, this is just the case where you’re running Grafana locally so we can early exit

Reply:
I'm ok with that, but I guess my question is how would we get here if running Grafana locally? Seems like something would be wrong and a user should be warned?

Reply:
Updated this to just handle the error as well.